EIPS/eip-7511.md

```diff
@@ -0,0 +1,222 @@
+| [04] | 5f     | PUSH0          | 0                  |
+| [05] | 5f     | PUSH0          | 0 0                |
+| [06] | 36     | CALLDATASIZE   | cds 0 0            |
+| [07] | 5f     | PUSH0          | 0 cds 0 0          |
+| [08] | 73bebe.| PUSH20 0xbebe. | 0xbebe. 0 cds 0 0  |
+| [1d] | 5a     | GAS            | gas 0xbebe. 0 cds 0 0|
+| [1e] | f4     | DELEGATECALL   | suc                |
+```
+
+### Step 3: Copy the Returned Data from the `DELEGATECALL`
+
+To copy the returndata, we need to provide the arguments for the `RETURNDATACOPY` opcodes, which are `[0, 0, red]`, where `rds` represents size of returndata from the `DELEGATECALL`.
+
+```shell
+| pc   | op     | opcode         | stack              |
+|------|--------|----------------|--------------------|
+| [1f] | 3d     | RETURNDATASIZE | rds suc            |
+| [20] | 5f     | PUSH0          | 0 rds suc          |
+| [21] | 5f     | PUSH0          | 0 0 rds suc        |
+| [22] | 3e     | RETURNDATACOPY | suc                |
+```
+
+### Step 4: Return or Revert
+
+Lastly we need to return the data or revert the transaction based on whether the `DELEGATECALL` is successful. There is no `if/else` in opcodes, so we need to use `JUMPI` and `JUMPDEST` instead. The auguments for `JUMPI` is `[0x2a, suc]`, where `0x2a` is the destination of the conditional jump.
+
+ We also need to prepare the argument `[0, rds]` for `REVERT` and `RETURN` opcodes before the `JUMPI`, otherwise we have to prepare them twice. We cannot avoid the `SWAP` operation, because we can only get `rds` after the `DELEGATECALL`.
+
+```shell
+| pc   | op     | opcode         | stack              |
+|------|--------|----------------|--------------------|
+| [23] | 5f     | PUSH0          | 0 suc              |
+| [24] | 3d     | RETURNDATASIZE | rds 0 suc          |
+| [25] | 91     | SWAP2          | suc 0 rds          |
+| [26] | 602a   | PUSH1 0x2a     | 0x2a suc 0 rds     |
+| [27] | 57     | JUMPI          | 0 rds              |
+| [29] | fd     | REVERT         |                    |
+| [2a] | 5b     | JUMPDEST       | 0 rds              |
+| [2b] | f3     | RETURN         |                    |
+```
+
+In the end, we arrived at the runtime code for Minimal Proxy Contract with `PUSH0`: 
+
+```
+365f5f375f5f365f73bebebebebebebebebebebebebebebebebebebebe5af43d5f5f3e5f3d91602a57fd5bf3
+```
+
+The length of the runtime code is `44` bytes, which reduced `1` byte from the previous Minimal Proxy Contract. Moreover, it replaced the `RETURNDATASIZE` and `DUP` operations with `PUSH0`, which saves gas and increase the readability of the code. In summary, the new Minimal Proxy Contract reduce `200` gas at deployment and `5` gas at runtime, while remaining same functionalities as the old one.
+
+## Backwards Compatibility
+
+Because the new minimal proxy contract uses `PUSH0` opcode, it can only be deployed after Shanghai Upgrade. It behaves the same as previous Minimal Proxy Contract.
```

Comment:
```suggestion
Because the new minimal proxy contract uses `PUSH0` opcode, it can only be deployed after Shanghai Upgrade. It behaves the same as the previous Minimal Proxy Contract.
```